File: projects/003-url-shortener-microservice/solutions/robertozecchini/url_shortener.py
#!/usr/bin/env python3

from flask import Flask, jsonify, request, redirect
import url_collection

app = Flask(__name__)
urls = url_collection.UrlCollection()

@app.route('/api/shorturl', methods = ['POST'])
def shorturl():
    raw_data = request.get_data()
    shorturl = urls.add_url(raw_data.decode('UTF-8'))
    if shorturl == 0:                   #invalid url
        return jsonify({'error': 'invalid url'})
    else:
        return jsonify({'original_url': str(urls.get_original(shorturl)), 'short_url': shorturl})

@app.route('/api/shorturl/<short_url>')
def redirect_shorturl(short_url):
    try:
        shorturl = int(short_url)
        url = urls.get_original(shorturl)
        if url == '':
            raise ValueError
        else:
            return redirect(url)
    except ValueError:
        return jsonify({'error': 'invalid short url'})

def clear_collection():
    urls.clear()


if __name__ == '__main__':
    app.run(host='0.0.0.0', debug=True) 

File: projects/004-url-shortener-microservice/solution/Davide Di giacomo/urlShortener.js
const express = require('express')
const app = express()
const {url} = require("./url")


//POST a URL to /api/shorturl and get a JSON response with original_url and short_url properties.
app.post("/api/shorturl", (req, res) => {
    res.status(200).json({url});
})

//URL to /api/shorturl and get a JSON response with original_url and short_url properties.
app.get("/api/shorturl/:short_url", (req, res) => {

    const { short_url } = req.params

    for (let i=0; i<url.length; i++) {
        if (url[i].short_url === Number(short_url)) {
            res.redirect(url[i].original_url);
        }
    }
    res.json({success: false, error: "Invalid Url"})

})

//the JSON response will contain { error: 'invalid url'}
app.get("*", (req, res) => {
    res.json({success: false, error: "Invalid Url"})
})


app.listen(3000)